### DISCUSSION TITLE
Pdf e markdown in dynamik

### DISCUSSION THREAD

**Discussion by samuelemusiani** [csunibo/csunibo.github.io#47] [Sondaggi]
Perdonate il nome ma non sapevo che scegliere. La questione sarebbe se mantenere sia i markdown che i pdf compilati (relativi ai markdown) visibili all'utente in dynamik. Ovviamente dynamik non c'entra perché sono scelte che facciamo a livello di CD/Ci quando con statik scegliamo quali file andranno nell'indicizzazione finale.

La scelta andrebbe poi mantenuta consistente tra tutte le repo.

**Comment by Flecart** [comment]
Per me è una buona idea.
Al massimo se una cartella diventa troppo popolata a causa di questo si possono mettere filtri lato dynamic su localstorage

**Comment by samuelemusiani** [comment]
Secondo me crea un po' di confusione averli entrambi nella stessa pagina. Perché avendo 10 markdown in una cartella i file diventa 20 e tutti duplicati. Ha senso tenerli se su dynamik ci fosse una logica per non mostrarli tutti ma mostrarli "raggruppati" e lasciamo la scelta finale all'utente se aprire il pdf o il markdown. Basterebbe prendere i file con lo stesso nome e mostrarne solo uno come predefinito, avendo però una tendina (?) al lato che permette di selezionare la versione/estensione che si preferisce. Inoltre si potrebbe mettere nelle impostazioni la preferenza per il formato che si vuole visualizzare subito. @csunibo/frontend

**Reply by ali-benny** [reply]
sì eh l'anno scorso avevo proposto la csunibo/dynamik#1 e potremmo fare qualcosa di simile (?) anche per questo caso: magari nella pagina dedicata al file mettere delle tabs [ md | pdf | ... ] https://flowbite.com/docs/components/tabs/

**Reply by foxyseta** [reply]
Questo significa un click in più però, e il dovere avere un'interfaccia anche nella pagina della risorsa.
Se invece dynamik mostrasse questi pulsanti a fianco al nome del file nell'elenco?

> appunti-di-clara [link] [md] [pdf] 
> appunti-di-giacomo [link] [md] [pdf]

In altre parole lato static non cambia un tubo (il suo json non lo tocchiamo), ma dynamik raggruppa nella stessa riga due file quando hanno lo stesso nome, anche se hanno estensioni diverse. E se per qualche motivo le date differiscono si può mettere un tooltip.

EDIT: OP @samuelemusiani ha aperto https://github.com/cartabinaria/dynamik/issues/289

**Comment by foxyseta** [comment]
Questo discorso vogliamo applicarlo anche a LaTeX + PDF, Typst + PDF e Asciidoc + HTML? Scusate il commento sulla discussione chiusa ma mi pare sia una cosa da decidere.

**Reply by samuelemusiani** [reply]
Volendo non sarebbe male. Più che altro bisogna scrive il codice ahahah

**Reply by foxyseta** [reply]
Sisi giusto per capire se scrivere codice generalizzato sarebbe inutile o no.

A una certa si parlava di compilare in HTML pure le cose che al momento sono compilate in pdf?

**Reply by samuelemusiani** [reply]
Dici i markdown per esempio?
